### DISCUSSION TITLE
Prepopulate (append fields to) a dynamic form

### DISCUSSION THREAD

**Discussion by mikekoro** [react-hook-form/react-hook-form#4511] [Q&A]
I am using MUI Autocomplete in conjunction with react-hook-form's useFieldArray to generate dynamic form fields. How can I prepopulate that form by using append?

I believe the culprit of it is `defaultValue={[]}` but what should I exactly set it to?

https://codesandbox.io/s/react-hook-form-field-array-advanced-with-delete-insert-append-edit-forked-srqmv?file=/src/index.js:0-2455

**Comment by bluebill1049** [comment]
can you try with `reset({...})` or `setValue('fieldArraynName', [])`

**Reply by mikekoro** [reply]
Well, I am trying to populate the form, not reset.
